platform/test/spec/testToolConfigValidatorSpec.js

```diff
@@ -0,0 +1,84 @@
+/*global describe, it, expect, beforeEach --  functions provided by Jasmine */
+import { ToolConfigValidator } from "../../src/ToolConfigValidator"
+import { TOOL_1PANELDEF_1FUNCTION } from "../resources/TestToolFiles.js";
+import { ConfigValidationError } from "../../src/ConfigValidationError.js"
+
+
+const EXPECTED_FILE_TYPE = "ToolConfig"; 
+
+describe("ToolConfigValidator", () => {
+      
+    describe("constructor", () => {
+        it("can be created", () => {
+            // Call the target object
+            let tcv = new ToolConfigValidator();
+            
+            // Check the expected results
+            expect(tcv).toBeInstanceOf(ToolConfigValidator);
+        })
+    })
+
+    describe("validate tool configration", () => {
+        let tcv;
+        let toolConfig;
+        
+        //Setup
+        beforeEach( () => {
+            toolConfig = JSON.parse(TOOL_1PANELDEF_1FUNCTION);
+            tcv = new ToolConfigValidator();
+        })
+
+        it("reports no errors for a valid config", () => {
+            // Call the target object
+            let errors = tcv.validateConfigFile(toolConfig);
+            
+            // Check the expected results
+            expect(errors).toHaveSize(0);
+        })
+
+        it("returns errors that are ConfigValidationError instances",() => {
+            delete toolConfig.tool.id;
+
+            // Call the target object
+            let errors = tcv.validateConfigFile(toolConfig);
+            let e = errors[0];
+            
+            // Check the expected results
+            expect(e).toBeInstanceOf(ConfigValidationError);
+            expect(e.fileType).toEqual(EXPECTED_FILE_TYPE);
+        })
+
+        it("returns an error if the config has no id key", () => {
```

Comment:
Merge with previous test.

Reply:
This is being check by `CheckErrorPopulated()` so removed. Fixed in cb677e9410c8154ce39faa02c77f690e60f89964